symengine/python/symengine/lib/symengine_wrapper.pyx

```diff
@@ -51,6 +47,32 @@ cdef c2py(RCP[const symengine.Basic] o):
         r = RealMPFR.__new__(RealMPFR)
     elif (symengine.is_a_ComplexMPC(deref(o))):
         r = ComplexMPC.__new__(ComplexMPC)
+    elif (symengine.is_a_Log(deref(o))):
+        r = Log.__new__(Log)
+    elif (symengine.is_a_Sin(deref(o))):
+        r = Sin.__new__(Sin)
+    elif (symengine.is_a_Cos(deref(o))):
+        r = Cos.__new__(Cos)
+    elif (symengine.is_a_Tan(deref(o))):
+        r = Tan.__new__(Tan)
+    elif (symengine.is_a_Cot(deref(o))):
+        r = Cot.__new__(Cot)
+    elif (symengine.is_a_Csc(deref(o))):
+        r = Csc.__new__(Csc)
+    elif (symengine.is_a_Sec(deref(o))):
+        r = Sec.__new__(Sec)
+    elif (symengine.is_a_ASin(deref(o))):
+        r = ASin.__new__(ASin)
+    elif (symengine.is_a_ACos(deref(o))):
+        r = ACos.__new__(ACos)
+    elif (symengine.is_a_ATan(deref(o))):
+        r = ATan.__new__(ATan)
+    elif (symengine.is_a_ACot(deref(o))):
+        r = ACot.__new__(ACot)
+    elif (symengine.is_a_ACsc(deref(o))):
+        r = ACsc.__new__(ACsc)
+    elif (symengine.is_a_ASec(deref(o))):
+        r = ASec.__new__(ASec)
```

Comment:
This is fine for this PR, but after it's merged, we should just read the type id and then just compare an integer, as opposed to calling all these functions.